packages/types/src/report.ts

```diff
@@ -1,59 +1,68 @@
-import { MappedRecordAttributes, RecordAttributes } from "@/utils";
+import { IFilter } from "@/index";
+import { Paginated } from "@/utils";
 
 export type Row = {
   id: number;
+  user_id: number | null;
   title: string;
   description: string;
-  category: string;
+  screenshot: string | null;
+  log: string | null;
   resolved: boolean;
-  resolved_at: Date | null;
   created_at: Date;
-  created_by: number;
   updated_at: Date;
-  updated_by: number;
 };
 
 export type Self = {
   id: number;
+  userId: number | null;
   title: string;
   description: string;
-  category: string;
+  screenshot: string | null;
+  log: string | null;
   resolved: boolean;
-  resolvedAt: string | null;
   createdAt: string;
-  createdBy: number;
   updatedAt: string;
-  updatedBy: number;
 };
 
-type SelectableSelf = Omit<
-  Self,
-  "createdAt" | "updatedAt" | "createdBy" | "updatedBy" | "resolvedAt"
->;
+export type CreatePayload = {
+  userId: number;
+  title: string;
+  description: string;
+  screenshot?: string;
+  log?: string;
+};
 
-export type Attributed = SelectableSelf &
-  RecordAttributes & {
-    resolvedAt: Date | null;
-  };
+export type UpdatePayload = {
+  resolved: boolean;
+};
 
-export type MappedAttributes = SelectableSelf &
-  MappedRecordAttributes & {
-    resolvedAt: string | null;
-  };
+export type FindModelQuery = IFilter.SkippablePagination & {
+  ids?: number[];
+  users?: number[];
+  title?: string;
+  description?: string;
+  screenshot?: boolean;
+  log?: boolean;
+  resolved?: boolean;
+  createdAt?: IFilter.Date;
+  updatedAt?: IFilter.Date;
+};
 
-export type CreatePayload = {
+export type CreateApiPayload = {
   title: string;
   description: string;
-  category: string;
-  createdBy: number;
+  /**
+   * An image encoded in base64
+   */
+  screenshot?: string;
+  log?: string;
```

Comment:
The image should be uploaded as a file `blob` or a `buffer`. This mean that the screenshot or the log file will not be included in body isntead will be included as a form file `req.files`. Check how the tutor assets are being uploaded.